dramatiq/brokers/rabbitmq.py

```diff
@@ -243,13 +246,13 @@ def declare_queue(self, queue_name, *, ensure=False):
             self.emit_after("declare_delay_queue", delayed_name)
 
         if ensure:
-            self._ensure_queue(queue_name)
+            self._ensure_queue(queue_name, ensure == "strict")
 
-    def _ensure_queue(self, queue_name):
+    def _ensure_queue(self, queue_name, strict=False):
         attempts = 1
         while True:
             try:
-                if queue_name in self.queues_pending:
+                if strict or queue_name in self.queues_pending:
```

Comment:
Taking another look at this, I'm not sure the strict argument is even required. Since the block at line 350 adds the missing queue to the `queues_pending` set, when the next message is enqueued, the queue will be created anyway, won't it? 

Separately, defaulting `ensure` to `"strict"` will cause the queue to be redeclared on every enqueue, which seems bad. If possible, I think we should drop the `strict` stuff altogether, or default `ensure` to `True` as before.

Reply:
The `ensure` value during enqueue is still `True`, the queue won't be declared actually every time util it's added to `queues_pending` set by some error handle logic.

I also move `self.declare_queue(queue_name, ensure=True)` to enqueue retry loop, make sure user won't retry again by themselves to redeclared the queue.

The `strict` is set only during consumer starting to ensure the queue is redeclared. 
Because there's no extra broker specific handle logic which can be implemented in consumer to handle with the redeclared of rabbitmq queue, and the consumer won't start/restart many time.

Reply:
Hello, @Bogdanp, could you take some time to review this response? Thanks to your patient and meticulous!